File: ROS/catkin_ws/src/my_robot_tutorials/src/scripts/smartphone.py
#!/usr/bin/env python3

import rospy
from std_msgs.msg import String

# creating a function to pass to subscriber node
def callback_recieve_radio_data(msg):
	rospy.loginfo("Message recieved: ")
	rospy.loginfo(msg)

if __name__ == '__main__':
	rospy.init_node('smartphone')

	# create a subscriber
	# arguments: topic name, topic type, the function to be performed using the topic data
	sub = rospy.Subscriber("/robot_news_radio", String, callback_recieve_radio_data)

	# spin will keep running the whole "smartphone" node code until it is stopped
	rospy.spin()

File: ROS/catkin_ws/src/my_robot_tutorials/src/smartphone.cpp
#include <ros/ros.h>
#include <std_msgs/String.h>

void callback_recieve_radio_data(const std_msgs::String &msg) {
	ROS_INFO("Message recieved: %s", msg.data.c_str());
}

int main(int argc, char **argv) {

	// initialize the node
	ros::init(argc, argv, "smartphone");
	// initialize NodeHandle
	ros::NodeHandle nh;

	// create a subscriber node using nodehandle object subscribe.
	// arguments: (topic name, buffer size, a function handling the recieved data)
	ros::Subscriber sub = nh.subscribe("/robot_news_radio", 1000, callback_recieve_radio_data);

	// it keeps the node running until we ask it to shutdown
	ros::spin();
}